drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -75,6 +84,47 @@ local child_device_profile_overrides = {
 
 local detect_matter_thing
 
+local START_BUTTON_PRESS = "__start_button_press"
+local TIMEOUT_THRESHOLD = 10 --arbitrary timeout
+local HELD_THRESHOLD = 1
+-- this is the number of buttons for which we have a static profile already made
+local STATIC_PROFILE_SUPPORTED = {2, 3, 4, 5, 6, 7, 8}
+
+local DEFERRED_CONFIGURE = "__DEFERRED_CONFIGURE"
+
+-- Some switches will send a MultiPressComplete event as part of a long press sequence. Normally the driver will create a
+-- button capability event on receipt of MultiPressComplete, but in this case that would result in an extra event because
+-- the "held" capability event is generated when the LongPress event is received. The IGNORE_NEXT_MPC flag is used
+-- to tell the driver to ignore MultiPressComplete if it is received after a long press to avoid this extra event.
+local IGNORE_NEXT_MPC = "__ignore_next_mpc"
+
+-- These are essentially storing the supported features of a given endpoint
+-- TODO: add an is_feature_supported_for_endpoint function to matter.device that takes an endpoint
+local EMULATE_HELD = "__emulate_held" -- for non-MSR (MomentarySwitchRelease) devices we can emulate this on the software side
+local SUPPORTS_MULTI_PRESS = "__multi_button" -- for MSM devices (MomentarySwitchMultiPress), create an event on receipt of MultiPressComplete
+local INITIAL_PRESS_ONLY = "__initial_press_only" -- for devices that support MS (MomentarySwitch), but not MSR (MomentarySwitchRelease)
+
+local HUE_MANUFACTURER_ID = 0x100B
+
+--helper function to create list of multi press values
+local function create_multi_list(size, supportsHeld)
+  local list = {"pushed", "double"}
```

Comment:
can we get a more clear name? I think in general helper functions should support as high a level of clarity as anything else

Reply:
Updated to `create_multi_press_values_list` in de641ad73df7965367fa355d0b54a3570ccec054